src/index.ts

```diff
@@ -117,14 +132,21 @@ const withSpidAuthMiddleware = (
  * Apply SPID authentication middleware
  * to an express application.
  */
+// tslint:disable-next-line: parameters-max-number
 export function withSpid(
   appConfig: IApplicationConfig,
   samlConfig: SamlConfig,
   serviceProviderConfig: IServiceProviderConfig,
   redisClient: RedisClient,
   app: express.Express,
   acs: AssertionConsumerServiceT,
-  logout: LogoutT
+  logout: LogoutT,
+  logCallback?: (
+    sourceIp: string | null,
+    payload: string,
+    timestamp: string,
+    isRequest: boolean
+  ) => void
```

Comment:
can you make a type for this method ie. `WithSpidCallbackT` and export it ?
(so the caller can use the exported type if she wants to define the callback outside withspid())